src/scenic/domains/driving/roads.py

```diff
@@ -57,7 +57,7 @@ def _toVector(thing: Vectorlike) -> Vector:
 
 
 def _rejectSample(message):
-    if veneer.isActive():
+    if not veneer.allowSampleRejection():
```

Comment:
Since `allowSampleRejection` should only ever be used in this way, I'd suggest replacing it with a function `veneer.rejectSample` which copies the logic of this function (raising either `InvalidScenarioError` or `RejectionException` as appopriate). Then the roads library and other libraries can call `veneer.rejectSample` when they detect a requirement violation and not worry about how to handle it.

Reply:
Good idea, I'll make the swap!